azurerm/internal/services/web/registration.go

```diff
@@ -52,3 +53,18 @@ func (r Registration) SupportedResources() map[string]*schema.Resource {
 		"azurerm_function_app_slot":                                 resourceFunctionAppSlot(),
 	}
 }
+
+// PackagePath is the relative path to this package
+func (r Registration) PackagePath() string {
+	return "TODO: do we need this?"
+}
+
+func (r Registration) DataSources() []sdk.DataSource {
+	return []sdk.DataSource{}
```

Comment:
The existing non-versioned data source works today, but since that's not guaranteed to remain the case throughout the Preview and contains response data that may be misleading, I'll add a specific one for V3.